src/T2DMap.cpp

```diff
@@ -4033,14 +4033,28 @@ void T2DMap::slot_setArea()
     }
 
     QMapIterator<int, QString> it(mpMap->mpRoomDB->getAreaNamesMap());
-    while (it.hasNext()) {
+    QStringList sortedAreaList;
+    QStringList::const_iterator constAreaIterator;
+    sortedAreaList = mpMap->mpRoomDB->getAreaNamesMap().values();
+
+    while (it.hasNext()){
         it.next();
-        int areaID = it.key();
-        if (areaID > 0) {
-            arealist_combobox->addItem(QStringLiteral("%1 (%2)").arg(it.value(), QString::number(areaID)), QVariant(areaID));
+        if (it.key() > 0){
+            sortedAreaList += it.value();
```

Comment:
Isn't `sortedAreaList` a list of area names already? So no need to iterate `getAreaNamesMap()` again and add the areas, right?

Reply:
let me check

Reply:
fixed. Thanks, Vadi!